Runtime/Scripts/BaseMeshSync.cs

```diff
@@ -712,13 +712,72 @@ internal void AfterUpdateScene()
             {
                 // mark scene dirty
                 EditorSceneManager.MarkSceneDirty(SceneManager.GetActiveScene());
+
+                UpdateLODGroups();
             }
 #endif
 
             if (onSceneUpdateEnd != null)
                 onSceneUpdateEnd.Invoke();
         }
 
+        void UpdateLODGroups() {
+            if (!GenerateLODGroup) {
+                return;
+            }
+
+            // Add an LOD group component if we have LODs:
+
+            // Find out how many LODs we got:
+            int lodCount = 0;
+
+            Dictionary<int, List<Renderer>> lodRenderers = new Dictionary<int, List<Renderer>>();
+
+            foreach (var child in GetComponentsInChildren<Transform>()) {
+                // If the name ends in LODx it needs to be put in an LOD group:
+                if (child.name.IndexOf("LOD") == child.name.Length - 4) {
+                    if (int.TryParse(child.name.Substring(child.name.Length - 1), out int lod)) {
```

Comment:
can you invert these if statements?

Reply:
You could also use a regex here, something like "(any char)LOD(any num, any times)". This would remove the need for a TryParse (just Parse).

What is the max num of LODs supported in the Engine? if it goes above 9, this would not work properly.

Reply:
Unity's LOD group supports a maximum of 8 LODs. I deliberately avoided regex because it's slower.